app/main.py

```
@@ -1 +1,47 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        if isinstance(skill, str):
+            self.skills.append(skill)
+        elif isinstance(skill, list):
+            self.skills.extend(skill)
```

Checklist issue: The `learn_skill` method should handle both a single string and a list of strings, but it does not handle invalid input types. According to the checklist, you should add an `else` clause to raise an exception or handle unexpected input types.